.nix-helpers/termonad-with-packages.nix

```diff
@@ -23,13 +23,16 @@
 
 let
   nixpkgs  = import ./nixpkgs.nix;
-  termonad = nixpkgs.callPackage ../. {};
-  defghcWithP = nixpkgs.haskellPackages.ghcWithPackages;
   defpackages = self: [ self.colour self.lens ];
 in
 
-{ ghcWithPackages ? defghcWithP, packages ? defpackages }:
-let env = ghcWithPackages (self: [ termonad ] ++ packages self); in
+{ packages ? defpackages, compiler ? "ghc843" }:
+
+let
+  ghcWithPackages = nixpkgs.haskell.packages."${compiler}".ghcWithPackages;
+  termonad = import ./bare.nix { inherit compiler; };
+  env = ghcWithPackages (self: [ termonad ] ++ packages self);
+in
 
 { stdenv, makeWrapper }:
```

Comment:
Is there some reason that stdenv and makeWrapper aren't being obtained from `nixpkgs` on line 25, but instead left to be passed in.

It seems like the code could be made a little simpler by just taking these from `nixpkgs`, so I'm wondering if we gain anything by not doing that?

Reply:
I think you're right; if there's a perfectly good `nixpkgs` in scope there's no need for those arguments. I'm just used to writing derivations in this `callPackage` style and did so automatically, but simplifying them out would also make the choice of `callPackage` in `default.nix` moot.